Felipe - felipegermanos/Week 10/fgc-week_10.do

```diff
@@ -0,0 +1,191 @@
+merge m:1 department using `density'
+
+
+/* Finally, we drop empty rows (case when merge ==2) */
+drop if _merge==2 
+drop _merge
+
+
+/* And this exports the dataset */
+save question_1, replace
+
+
+
+
+
+********************************************************************************
+******************************   Question 2   **********************************
+********************************************************************************
+
+/* We also covered most of this algorythm in class - ordering latitude and longitude
+and choosing appropriate clusters following such ordering. Notice that we need to 
+use a package, geodist, to run this code, so there might be an error because of it */
+
+
+use "$gps_data", clear
+
+
+/* We start by generating a scatterplot to check for clusters of observations */
+sort latitude longitude
+scatter longitude latitude, title("Q2 - Clusters by latitude and longitude")
+
+/* Next, we divide the data, which is sorted, in blocks of six, to reflect
+the fact that each field worker is assigned to 6 households, and use geodist to 
+check how efficient our ordering was. Notice nothing in the code is specific to 
+this dataset, so it should be generalizible */
+egen field_worker = seq(), block(6) 
+bysort field_worker (longitude latitude): gen lat_shifted = latitude[_n+1]
+bysort field_worker (longitude latitude): gen longit_shifted = longitude[_n+1] 
+geodist latitude longitude lat_shifted longit_shifted, generate(distance)
+
+/* Finally, we can try to measure how well we did by summing the total distance
+travelled */
+
+bysort field_worker: egen distance_travelled = total(distance)  
+sum distance_travelled 
+
+
+tabulate distance_travelled // all enumerators are walking under 2 miles
+summarize distance_travelled // enumerators are walking under 1 mile on average
+
+
+/* And this exports the dataset */
+save question_2, replace
+
+
+
+
+
+********************************************************************************
+******************************   Question 4   **********************************
+********************************************************************************
+
+
+
+use "$tz_elec_10_clean", clear
+
+/* Data cleaning to ensure smoother merge */
+rename (region_10 district_10 ward_10) (region district ward)
+drop total_can ward_tot
+
+/* And we are ready to save the temporary file*/
+tempfile old_ward
+save `old_ward', replace
+
+
+/* Here is the code we use to merge the datasets on "district." Notice, we also
+have to perform some analogous cleaning work */
+use "$tz_elec_15_clean", clear
+rename (region_15 district_15 ward_15) (region district ward)
+drop total_can ward_tot
+merge 1:1 region district ward using `old_ward' 
+
+/* Final cleaning */
+
+gen ward_final = _merge
+drop _merge
+
+sort ward_final
+label var ward_final "Ward"
+label def ward_final 1 "parentless" 2 "childless" 3 "full"
+
+tabulate ward_final
+
+
+/* Exporting results */
+save question_3, replace

```

Comment:
I basically did the same process for my code, but I worry that it's an oversimplification of the problem. In particular, it seems that wards could be in different regions/districts in 2010 and 2015, and this affects the validity of our merge.

Something to take a look at is that if you run "sort ward ward_final" at the end of this process, you can quickly find that there are many instances where two wards have the same name, and one is parentless and one is childless. I think this seems suspicious, especially when the wards are in the same region OR district (but not both)! For instance, observations 14 and 15 are both the ward "ayamaami" from the region "manyara", but the parentless ward is from the district "mji wa mbulu" and the childless ward is from the district "wilaya ya mbulu." It seems plausible that redistricting happened here and that these are the same ward.

One idea I had was to merge only using region and ward (to avoid the redistricting issue), but you can't merge only using the variables region and ward because they "do not uniquely identify observations in the master data." 

I wish I could be more helpful but I still haven't solved this problem myself. I'd love to hear your thoughts on this.

Reply:
I have some updates on this after editing my own submission.

If you run "tab region _merge", or "tab region ward_final" in your specific case, you'll see that 4 regions only have parentless observations. This means these regions were created between 2010 and 2015. However, many of the ward names remain the same despite being placed in a new region and district. If you tag the observations from new regions (with an indicator variable), you can use that tag and ward names to figure out whether you're dealing with the same region even if the region and district names are different.

Another aspect here is that many of the wards are in the same region in 2010 and 2015, but a different district. You can sort by ward, region, and _merge (or ward_final) to tag these observations that are the same ward but just redistricted.

It was quite a long process for me and is difficult to explain, but feel free to look at my submission if you're interested. Let me know if you think I am overthinking the problem or if you have any questions.